util-cache/src/main/scala/sbt/util/CacheStore.scala

```diff
@@ -47,38 +49,44 @@ abstract class CacheStoreFactory {
 }
 
 object CacheStoreFactory {
+  @deprecated("Create your own IsoString[JValue]", "1.4")
   implicit lazy val jvalueIsoString: IsoString[JValue] =
     IsoString.iso(CompactPrinter.apply, Parser.parseUnsafe)
 
   /** Returns directory-based CacheStoreFactory using standard JSON converter. */
   def apply(base: File): CacheStoreFactory = directory(base)
 
   /** Returns directory-based CacheStoreFactory using standard JSON converter. */
-  def directory(base: File): CacheStoreFactory = new DirectoryStoreFactory[JValue](base, Converter)
+  def directory(base: File): CacheStoreFactory = new DirectoryStoreFactory(base)
 }
 
 /** A factory that creates new stores persisted in `base`. */
-class DirectoryStoreFactory[J: IsoString](base: File, converter: SupportConverter[J])
-    extends CacheStoreFactory {
+class DirectoryStoreFactory[J](base: File) extends CacheStoreFactory {
   IO.createDirectory(base)
 
-  def make(identifier: String): CacheStore = new FileBasedStore(base / identifier, converter)
+  @deprecated("Use constructor without converter", "1.4")
+  def this(base: File, converter: sjsonnew.SupportConverter[J], isoString: sjsonnew.IsoString[J]) =
```

Comment:
this constructor was added, just for bincompt, but some parameters will just be ignored...